src/common/utils/get_resolved_document_by_id.py

```diff
@@ -58,8 +58,8 @@ def resolve_complete_document(entity, data_source, depth, depth_count) -> dict:
             if isinstance(value, list):  # If it's a list, resolve any references
                 entity[key] = resolve_reference_list(value, data_source, depth, depth_count)
             else:
-                if ref_id := value.get("_id"):  # It's a reference
-                    entity[key] = get_complete_sys_document(ref_id, data_source, depth_count)
+                if ref := value.get("ref"):  # It's a reference
```

Comment:
Same.
Could propably have a helper function to check if dict is reference.
Return type of reference (storage/link), or None if not a reference.